File: BookLand_FE/src/types/Notification.ts
export type NotificationStatus = 'UNREAD' | 'READ' | 'ARCHIVED';

export interface Notification {
    id: number;
    fromUserId?: number;
    fromUsername?: string;
    toUserId: number;
    toUsername: string;
    type: string;
    title: string;
    content: string;
    status: NotificationStatus;
    readAt?: string;
    createdAt: string;
}

File: services/notification-service/src/main/java/com/bookland/notification/entity/Notification.java
package com.bookland.notification.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    private String id;

    private String fromUserId;
    private String fromUsername;

    private String toUserId;
    private String toUsername;

    private String type;
    private String title;
    private String content;

    @Builder.Default
    private NotificationStatus status = NotificationStatus.UNREAD;

    private LocalDateTime readAt;
    private LocalDateTime createdAt;

    public enum NotificationStatus {
        UNREAD, READ, ARCHIVED
    }
}
